tcbuilder/backend/secure_boot.py

```
@@ -0,0 +1,644 @@
+    kernel_fitimage_path = check_if_file_exists(KERNEL_FITIMAGE, input_binaries_dir)
+
+    try:
+        # Check if mkimage is present in the environment
+        mkimage_version = subprocess.check_output(
+            [f"{UBOOT_TOOLS_DIR}/mkimage", "--version"],
+            text=True, stderr=subprocess.STDOUT)
+        log.info(f"Signing kernel fitImage with {mkimage_version}")
+
+        mkimage_env = {"SOURCE_DATE_EPOCH": "0"}
+        mkimage_cmd = [f"{UBOOT_TOOLS_DIR}/mkimage", "-D", "-I dts -O dtb -p 2000", "-F",
+                       "-k", kernel_key_dir, "-K", uboot_dtb_path, "-r", kernel_fitimage_path]
+
+        mkimage_print = (" ".join([f"{key}={val}" for key, val in mkimage_env.items()]) + " " +
+                         " ".join(mkimage_cmd))
+        log.info(mkimage_print)
+        print()
+        mkimage_output = subprocess.check_output(mkimage_cmd, text=True, env=mkimage_env,
+                                                 stderr=subprocess.STDOUT)
+
+    except subprocess.CalledProcessError as exc:
+        raise TorizonCoreBuilderError(exc.output.strip())
+
+    log.debug("---------- OUTPUT FROM MKIMAGE ----------")
+    log.debug(mkimage_output)
+    log.debug("--------- END OF MKIMAGE OUTPUT ---------")
+
+    wrong_keyname_match = re.search(r"Couldn't open RSA private key: .*/(\w+).key': No such file",
+                                    mkimage_output, re.IGNORECASE)
+
+    if wrong_keyname_match:
+        expected_keyname = wrong_keyname_match.group(1)
+        raise FileContentMissing(f"mkimage expected a key pair named ({expected_keyname}.key, "
+                                 f"{expected_keyname}.crt) to sign the kernel fitImage "
+                                 "(key-name-hint property). Aborting.")
+
+    sig_match = re.search(r"Signature written to '(.*)'", mkimage_output, re.IGNORECASE)
+    pub_match = re.search(r"Public key written to '(.*)'", mkimage_output, re.IGNORECASE)
+
+    if sig_match and pub_match:
```

I assume this if doesn't need an else branch because it would be impossible for that to happen?

---

Hmm, actually I forgot to consider this case when I wrote the function. I think the code should raise an exception there, just to be safe. I'll add it, thanks.